nnef/src/deser.rs

```diff
@@ -90,16 +102,27 @@ impl<'mb> ModelBuilder<'mb> {
             for (id, outlet) in identifiers.iter().zip(values.iter()) {
                 self.scopes.last_mut().unwrap().insert(id.to_string(), Value::Wire(*outlet));
             }
+            for (qparam, value) in datum_types.into_iter().zip(values.iter()) {
+                if let Some(qparam) = qparam {
+                    let node = self.model.node_mut(value.node);
+                    let output_fact = &mut node.outputs[value.slot].fact;
+                    output_fact.datum_type = qparam;
+                }
+            }
             self.naming_scopes.pop();
         }
         Ok(())
     }
 
-    pub fn wire_invocation(&mut self, invocation: &Invocation) -> TractResult<Value> {
+    pub fn wire_invocation(
+        &mut self,
+        invocation: &Invocation,
+        dt: &[Option<DatumType>],
```

Comment:
This is the datum type of the variable this invocation is assigned to. 
It is a slice of options to handle arrays and tuples of possibly unspecified types.